rti_connext_dds_cmake_module/cmake/rti_build_helper.cmake

```diff
@@ -529,18 +526,27 @@ function(rti_guess_connextdds_arch)
       if(CONNEXTDDS_ARCH)
         break()
       endif()
+
     endforeach()
-  endif()
 
-  if(NOT CONNEXTDDS_ARCH)
-    message(WARNING
-      "CONNEXTDDS_ARCH not specified. Please set "
-      "-DCONNEXTDDS_ARCH= to specify your RTI Connext DDS "
-      " architecture")
-  else()
+    if(NOT CONNEXTDDS_ARCH)
+      list(GET architecture_candidates 0 CONNEXTDDS_ARCH)
+      if(CONNEXTDDS_ARCH)
+        message(STATUS
+          "unsupported CMAKE_HOST_SYSTEM_NAME (${CMAKE_HOST_SYSTEM_NAME}) "
+          "or CMAKE_HOST_SYSTEM_PROCESSOR (${CMAKE_HOST_SYSTEM_PROCESSOR}). "
+          "Using architecture ${CONNEXTDDS_ARCH} anyway.")
+      else()
+        message(WARNING
+          "CONNEXTDDS_ARCH not specified. Please set "
+          "-DCONNEXTDDS_ARCH= to specify your RTI Connext DDS "
+          " architecture")
+      endif()
+    else()
       message(STATUS "Selected CONNEXTDDS_ARCH: ${CONNEXTDDS_ARCH}")
```

Comment:
I am afraid that this status message could not be output if the [guessed_architecture](https://github.com/ros2/rmw_connextdds/blob/609f3ef699757b2305ea129b236b318415b41689/rti_connext_dds_cmake_module/cmake/rti_build_helper.cmake#L463) is matched, but the original logic did.

Reply:
indeed, i modified the message in the case the guessed_architecture exists, please refer to below